### PR TITLE
HOCS 2619 Serenity Automation Drone Changes

### PR DESCRIPTION
**Pull request by Paulcr-ho**
Converting for drone 1.0

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -99,6 +76,25 @@ steps:
     - success
     - failure
 
+---
+
```
Comment (zuzak) [inline .drone.yml:80]: I don't think we need this? Although a `kube-hocs-serenity-automation` repo exists, we don't seem to use it? We don't need to host the app in our environments as a pod. If we don't need it, remove all of this green chunk.

File: .drone.yml [hunk 2]
```diff
@@ -8,53 +8,30 @@ platform:
...
 
-- name: image_to_quay
-  pull: if-not-exists
-  image: quay.io/ukhomeofficedigital/drone-docker
+- name: build & push
+  image: plugins/docker
   settings:
     registry: quay.io
-    repo: quay.io/ukhomeofficedigital/hocs-serenity-automation
+    repo: quay.io/ukhomeofficedigital/hocs_serenity_automation
```
Comment (zuzak) [inline .drone.yml:23]: This should remain the old name. We're changing the tags so the Drone 1.0 tags (which start at 1) don't overwrite the old 0.8 ones -- but the repo can stay the same.
```suggestion
    repo: quay.io/ukhomeofficedigital/hocs-serenity-automation
```

File: .drone.yml [hunk 3]
```diff
@@ -8,53 +8,30 @@ platform:
...
-  - docker push quay.io/ukhomeofficedigital/hocs-serenity-automation:$${DRONE_TAG}
-  environment:
-    DOCKER_HOST: tcp://172.17.0.1:2375
-    DOCKER_PASSWORD:
-      from_secret: docker_password
-  when:
-    event:
-    - tag
+      - push
+  depends_on:
```
Comment (zuzak) [inline .drone.yml:33]: I'd leave out the `depends_on` stuff for now. If it's defined, Drone will do a cool acyclic dependency graph thing, but if it's not it'll just run through the steps sequentially top to bottom. While you're still trying to figure out why everything's broken it'll just add complexity.

File: .drone.yml [hunk 4]
```diff
@@ -99,6 +76,25 @@ steps:
     - success
     - failure
 
+---
```
Comment (zuzak) [inline .drone.yml:79]: The service definitions need to be in the same pipeline as the build steps that consume them (i.e. above this `---`).
(A pipeline here meaning a discrete list of steps that run on a Drone build -- I know it's an overloaded term)

File: .drone.yml [hunk 5]
```diff
@@ -8,53 +8,30 @@ platform:
...
+- name: build & push
+  image: plugins/docker
   settings:
     registry: quay.io
-    repo: quay.io/ukhomeofficedigital/hocs-serenity-automation
+    repo: quay.io/ukhomeofficedigital/hocs_serenity_automation
     tags:
-    - build_${DRONE_BUILD_NUMBER}
-    - latest
+      - build_${DRONE_BUILD_NUMBER}
```
Comment (zuzak) [inline .drone.yml:25]: while we're here:
```suggestion
      - build_${DRONE_BUILD_NUMBER}
      - ${DRONE_COMMIT_SHA}
```

File: .drone.yml (hunk 1 above)
Comment (Paulcr-ho) [reply .drone.yml]: Removed

File: .drone.yml (hunk 3 above)
Comment (Paulcr-ho) [reply .drone.yml]: Removed.

File: .drone.yml (hunk 4 above)
Comment (Paulcr-ho) [reply .drone.yml]: Removed whole section for kube project.

**Review by zuzak — APPROVED** [review]
This will push on every commit as opposed to only on a push to master. That's fine though

**Comment by zuzak** [comment]
(remember to squash)

File: .drone.yml [hunk 6]
```diff
@@ -8,53 +8,29 @@ platform:
...
-  environment:
-    DOCKER_HOST: tcp://172.17.0.1:2375
-  when:
-    branch:
-    - master
-    event:
-    - push
+- name: build project
+    image: quay.io/ukhomeofficedigital/openjdk11:v11.0.5_10
+    commands:
```
Comment (zuzak) [inline .drone.yml:13]: This block needs to go in by two spaces. YAML is very picky about indentation.

Comment (Paulcr-ho) [reply .drone.yml]: Its COBOL all over again. Done.

**Comment by zuzak** [comment]
I don't think the dnd step's required anymore.

File: .drone.yml [hunk 7]
```diff
@@ -8,53 +8,40 @@ platform:
...
-  - docker login -u="ukhomeofficedigital+hocs_quay_robot" -p=$${DOCKER_PASSWORD} quay.io
-  - docker tag hocs-serenity-automation quay.io/ukhomeofficedigital/hocs-serenity-automation:$${DRONE_TAG}
-  - docker push quay.io/ukhomeofficedigital/hocs-serenity-automation:$${DRONE_TAG}
+- name: build & push latest
+  image: plugins/docker
+  settings:
+    registry: quay.io
+    repo: quay.io/ukhomeofficedigital/hocs-serenity-automation
+    tags:
+      - master
```
Comment (zuzak) [inline .drone.yml:33]:
```suggestion
      - latest
```

File: .drone.yml [hunk 8]
```diff
@@ -8,53 +8,40 @@ platform:
...
+      from_secret: QUAY_ROBOT_PASSWORD
+    DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
   when:
+    branch:
+      - master
     event:
-    - tag
+      - push
+  depends_on:
+    - build project
```
Comment (zuzak) [inline .drone.yml:44]: `build project` doesn't exist anymore so Drone won't understand what to do

Comment (Paulcr-ho) [reply .drone.yml]: removed.

**Pull request merged**